### DISCUSSION TITLE
Sway/Bspwm bundle

### DISCUSSION THREAD

**Discussion by dedguy21** [clearfraction/distribution#8] [Ideas]
**BSPWM**
----------------------------------------
- [x] [Bspwm](https://github.com/clearlinux-pkgs/bspwm) 
- [x] Kitty | alacritty
- [ ] Redshift - will be crazy painful to get it working from /opt 'cause geoclue needs dbus service & systemd service
----------------------------------------
- [x] Sxhkd
- [ ] Picom -- no screen tearing
- [x] Gvfs
- [x] PcmanFM
- [ ] Polybar
- [x] Xrandr
- [ ] Autorandr
- [x] feh 
      Lxappearance - updated in 2017, not worth it
      Lxsession - updated in 2014, not worth it

**SWAY**
----------------------------------------
- [x] Sway
- [x] SwayBG
- [x] Swayidle
- [x] Swaylock
- [x] Kitty | alacritty
- [x] rofi-lbonn-wayland-git | bemenu
- [x] wlsunset | wl-gammarelay - redshift alternative
----------------------------------------
- [x] Waybar  - no idea how to compile it, logs aren't helpful
- [x] Swaylock-effects (blurs lock screen)
- [x] Kanshi (multi monitor use)
- [x] Wlr-randr
- [x] Wl-clipboard
- [x] Autotiling
- [x] imv (image viewer)
- [x] Oguri - background setter
- [x] Swayblur - works with oguri to blur when terminal is active

Themes & cursors - no idea if they will work from /opt
---  - core/extras split

**Comment by dedguy21** [comment]
Thank you for looking into this. I'm in the process of setting up what I
would need to do to run a functional sway environment. Let you know what
pitfalls are with environment as is.

**Comment by paulcarroty** [comment]
@dedguy21 [added some packages](https://github.com/clearfraction/bundles/commit/9587caf0a11b9a046a8d17d9659b425cb6b0f7d3), you can extract their CI artifacts right now. Hope to add more before release.

**Comment by dedguy21** [comment]
No I prefer Wayland. Just the last time I used Nvidia on Sway it did not
work. And it was forced replaced with nouveau.

Thanks.

I haven't looked at the sway bundle but it definitely needs to have waybar
kanshi autotiling sysytemd-sway (if there's no modular session manager that
can be attached).

**Reply by paulcarroty** [reply]
@dedguy21 I've switched to Sway too, can't help with NVIDIA 'cause only Intel graphics around. Check the Sway wiki and probably you'll find some info.


> needs to have waybar kanshi autotiling

All provided by `sway-extras` and `waybar` [bundles](https://github.com/clearfraction/bundles/tree/master/configs).


> systemd-sway

Sway systemd service not provided 'cause manual setup will be required anyway, check [our wiki](https://github.com/clearfraction/distribution/wiki/Dbus-and-systemd-services).

Ah, we got the wiki^ and write access is granted for all, feel free to share something meaningful.

- https://github.com/swaywm/sway/wiki/Systemd-integration#start-sway-without-login-manager-like-gdmsddm
- https://github.com/swaywm/sway/wiki/Systemd-integration
- https://github.com/swaywm/sway/wiki#nvidia-users

**Comment by whoizit** [comment]
In **sway** I use [foot](https://codeberg.org/dnkl/foot) terminal as a direct alternative to **urxvt** for **Wayland**. This works in daemon mode (`exec foot --server`) and runs the terminals with the `footclient` command (`set $term footclient --no-wait`).
It uses 18.5 MiB RAM (Private + Shared) right now on my system with two opened terminals (see `sudo ps_mem` output)

Also [persway](https://github.com/johnae/persway) super powerful feature for **sway**. This adds the **bspwm**/**awesomewm** features to **sway**. I use it this way:
`exec persway daemon --workspace-renaming --default-layout spiral`

These packages are not in ClearLinux

**Reply by paulcarroty** [reply]
If you or anyone else would like to contribute - I'll add both.

**Reply by paulcarroty** [reply]
Wezterm will be available in the next release with server mode, multiplexing, images via sixel, ligatures, dynamic color schemes and [more](https://wezfurlong.org/wezterm/features.html#available-features).
